{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "rural-skating",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# HDL PRS example\n",
    "\n",
    "Here we show an example of our pipeline for HDL PRS on UK Biobank samples. We use both effects estimates from MVP lipid traits analysis as well as posterior effects generated by `mashr` package."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "documented-glasgow",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Data used"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7847ca0-e52d-41c7-89ec-ccc3387ebb63",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Reference panel\n",
    "\n",
    "Obtained via `download_1000G()` in `bigsnpr`. \n",
    "\n",
    "Including 503 (mostly unrelated) European individuals and ~1.7M SNPs in common with either HapMap3 or the UK Biobank. Classification of European population can be found at [IGSR](https://www.internationalgenome.org/category/population/). European individuals ID are from [IGSR data portal](https://www.internationalgenome.org/data-portal/sample).\n",
    "\n",
    "### GWAS summary statistics data\n",
    "\n",
    "From MVP. We have the original GWAS summary data as well as multivariate posterior estimate of HDL effects using [mashr](https://github.com/stephenslab/mashr). In brief, we have two versions of summary statistics (effect estimates) for HDL.\n",
    "\n",
    "### Target test data: UK biobank\n",
    "\n",
    "We select randomly from UK Biobank 2000 individuals with covariates and HDL phenotype (medication adjusted, inverse normalized). Their genotypes are extracted. See `UKB.QC.*` PLINK file bundle. \n",
    "\n",
    "### PRS Models\n",
    "\n",
    "Auto model runs the algorithm for 30 different $p$ (the proportion of causal variants) values range from 10e-4 to 0.9, and heritability $h^2$ from LD score regression as initial value.\n",
    "\n",
    "Grid model tries a grid of parameters $p$, ranges from 0 to 1 and three $h^2$ which are 0.7/1/1.4 times of initial $h^2$ estimated by LD score regression."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "accepting-pepper",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Analysis of MVP GWAS data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "rolled-ending",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 1: QC on reference panel\n",
    "\n",
    "Here we assume the target data QC has been already performed. We perform here QC for reference panel,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "israeli-costa",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "work_dir=mvp_gwas\n",
    "cd ~/Documents/PRS_MASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "stupid-preservation",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    },
    "kernel": "Markdown",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "sos run ldpred.ipynb snp_qc \\\n",
       "    --cwd $work_dir \\\n",
       "    --genoFiles 1000G.EUR/1000G.EUR.bed\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_qc \\\n",
    "    --cwd $work_dir \\\n",
    "    --genoFiles 1000G.EUR/1000G.EUR.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "foreign-analyst",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 2: Intersect SNPs among summary stats, reference panel and target data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4b00aa7b-7c22-4a54-b717-4b4f66bc1cc1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "work_dir=mvp_gwas\n",
    "lipid=tg\n",
    "data=gwas\n",
    "cd ~/Documents/PRS_MASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "invalid-channels",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32msnp_intersect_1\u001b[0m: SNP intersect of summary stats and genotype data\n",
      "INFO: \u001b[32msnp_intersect_1\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32msnp_intersect_1\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.intersect.rds /home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.intersect.snplist\u001b[0m\n",
      "INFO: Running \u001b[32msnp_intersect_2\u001b[0m: \n",
      "INFO: \u001b[32msnp_intersect_2\u001b[0m is \u001b[32mcompleted\u001b[0m (pending nested workflow).\n",
      "INFO: Running \u001b[32mpreprocess_1\u001b[0m: Filter SNPs and select individuals\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bed /home/surbut/Documents/PRS_MASH/mvp_gwas/ukbb_merged.5000_subset.snp_intersect.extracted.bed in 2 groups\u001b[0m\n",
      "INFO: Running \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m: \n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bk /home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.rds... (4 items in 2 groups)\u001b[0m\n",
      "INFO: \u001b[32msnp_intersect_2\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bed /home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.rds... (4 items)\u001b[0m\n",
      "INFO: Workflow snp_intersect (ID=w2fcf830a2337b6ec) is executed successfully with 4 completed steps and 6 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_intersect \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss mvpdata/$data\"_\"$lipid.rds \\\n",
    "    --genoFiles $work_dir/1000G.EUR.$work_dir.bed UKBB_broad/ukbb_merged.5000_subset.bed -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "committed-arrival",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"There are 448077 shared SNPs.\"\n"
     ]
    }
   ],
   "source": [
    "tail -1 $work_dir/$data\"_\"$lipid.intersect.stdout"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "integral-ranch",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 3: Harmonize alleles for shared SNPs\n",
    "\n",
    "To handle major/minor allele, strand flips and consequently possible flips in sign for summary statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "dried-mechanics",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32msnp_match\u001b[0m: \n",
      "INFO: \u001b[32msnp_match\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32msnp_match\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.snp_matched.rds /home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.snp_matched.snplist\u001b[0m\n",
      "INFO: Workflow snp_match (ID=wf804c0b6f6acbd48) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_match \\\n",
    "    --cwd $work_dir \\\n",
    "    --reference_geno $work_dir/1000G.EUR.$work_dir.snp_intersect.extracted.rds \\\n",
    "    --ss mvpdata/$data\"_\"$lipid.rds -s force"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "considered-infection",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 4: Calculate LD matrix and fit LDSC model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "several-armenia",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mldsc\u001b[0m: \n",
      "INFO: \u001b[32mldsc\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mldsc\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.snp_matched.ld.rds\u001b[0m\n",
      "INFO: Workflow ldsc (ID=we8eff4db65db3c9b) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb ldsc \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --reference-geno $work_dir/1000G.EUR.$work_dir.snp_intersect.extracted.rds -s force"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "returning-investigator",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 6: Estimate posterior effect sizes and PRS"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "political-stack",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "For original data,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "modular-midnight",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32minf_prs\u001b[0m: \n",
      "INFO: \u001b[32minf_prs\u001b[0m is \u001b[32mcompleted\u001b[0m (pending nested workflow).\n",
      "INFO: Running \u001b[32mprs_core\u001b[0m: \n",
      "INFO: \u001b[32mprs_core\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mprs_core\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: \u001b[32minf_prs\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_tg.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: Workflow inf_prs (ID=wd3dd1fe41a7c33a0) is executed successfully with 2 completed steps.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb inf_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data\"_\"$lipid.snp_matched.ld.rds -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "interpreted-temple",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"422921 SNPs are used for PRS calculations\"\n"
     ]
    }
   ],
   "source": [
    "tail -1 mvp_gwas/$data\"_\"$lipid.snp_matched.inf_prs.stdout"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "nonprofit-haven",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 7: predict phenotypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "female-uniform",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Baseline model: Traits ~ Sex + Age"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8804f629-e8ca-4737-8cd6-b02f57919280",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tg\n"
     ]
    }
   ],
   "source": [
    "echo $lipid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "pharmaceutical-cherry",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mpred_eval\u001b[0m: \n",
      "INFO: \u001b[32mpred_eval\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpred_eval\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/UKB.tg.baseline.rds\u001b[0m\n",
      "INFO: Workflow pred_eval (ID=wd5dfe822af5ca266) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fabulous-flour",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = ., data = dat[train.ind, ])\n",
       "\n",
       "Residuals:\n",
       "    Min      1Q  Median      3Q     Max \n",
       "-1.5856 -0.6484 -0.2391  0.4010  7.5583 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept) 1.100692   0.112827   9.756  < 2e-16 ***\n",
       "AGE         0.007420   0.001963   3.780 0.000159 ***\n",
       "SEX         0.473554   0.032322  14.651  < 2e-16 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 0.9918 on 3785 degrees of freedom\n",
       "Multiple R-squared:  0.0574,\tAdjusted R-squared:  0.0569 \n",
       "F-statistic: 115.2 on 2 and 3785 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>model</th><th scope=col>R2</th><th scope=col>MSE</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>model</td><td>0.0569</td><td>0.89706</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " model & R2 & MSE\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t model & 0.0569 & 0.89706\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| model &lt;chr&gt; | R2 &lt;dbl&gt; | MSE &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| model | 0.0569 | 0.89706 |\n",
       "\n"
      ],
      "text/plain": [
       "  model R2     MSE    \n",
       "1 model 0.0569 0.89706"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "setwd(\"~/Documents/PRS_MASH\")\n",
    "lipid=\"tg\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".baseline.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "different-tragedy",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Inf/grid/auto model: Traits ~ Sex + Age + PRS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "assumed-moment",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mpred_eval\u001b[0m: \n",
      "INFO: \u001b[32mpred_eval\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpred_eval\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/UKB.tg.gwas_tg.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: Workflow pred_eval (ID=wb8eeda1e8a2d713f) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/$data\"_\"$lipid.snp_matched.inf_prs.rds \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "sharing-report",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = ., data = dat[train.ind, ])\n",
       "\n",
       "Residuals:\n",
       "    Min      1Q  Median      3Q     Max \n",
       "-1.8820 -0.6095 -0.2046  0.3603  7.1212 \n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  1.259445   0.109326  11.520  < 2e-16 ***\n",
       "AGE          0.007267   0.001895   3.835 0.000127 ***\n",
       "SEX          0.469264   0.031201  15.040  < 2e-16 ***\n",
       "PRS         -0.828117   0.049661 -16.675  < 2e-16 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 0.9574 on 3784 degrees of freedom\n",
       "Multiple R-squared:  0.1219,\tAdjusted R-squared:  0.1212 \n",
       "F-statistic: 175.1 on 3 and 3784 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>model</th><th scope=col>R2</th><th scope=col>MSE</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>model.inf_prs</td><td>0.12123</td><td>0.8499</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " model & R2 & MSE\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t model.inf\\_prs & 0.12123 & 0.8499\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| model &lt;chr&gt; | R2 &lt;dbl&gt; | MSE &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| model.inf_prs | 0.12123 | 0.8499 |\n",
       "\n"
      ],
      "text/plain": [
       "  model         R2      MSE   \n",
       "1 model.inf_prs 0.12123 0.8499"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data=\"gwas\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.inf_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "faf4fd66-f325-46d5-b5c0-e0526f87a4ac",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Bonus steps: repeat Steps 6 and 7 using other PRS models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "923893ee-00f6-4a4b-81b8-594c1b39835d",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb grid_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data\"_\"$lipid.snp_matched.ld.rds \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6b0a4e8-c643-41f3-ad55-7bb77da8ffc4",
   "metadata": {
    "kernel": "Markdown",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb auto_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data\"_\"$lipid.snp_matched.ld.rds -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "least-cinema",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/$data\"_\"$lipid.snp_matched.grid_prs.rds \\\n",
    "    --phenoFile ukbiobank/UKB.$lipid.cov \\\n",
    "    --covFile ukbiobank/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fa95a97-824c-432e-9221-d52f227aecb3",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.grid_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "certified-marriage",
   "metadata": {
    "kernel": "Markdown",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/$data\"_\"$lipid.snp_matched.auto_prs.rds \\\n",
    "    --phenoFile ukbiobank/UKB.$lipid.cov \\\n",
    "    --covFile ukbiobank/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "regular-parking",
   "metadata": {
    "kernel": "Markdown",
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.auto_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "Markdown",
     "markdown",
     "markdown",
     "",
     "markdown"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.22.5"
  },
  "toc-autonumbering": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
