{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "rural-skating",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# HDL PRS example\n",
    "\n",
    "Here we show an example of our pipeline for HDL PRS on UK Biobank samples. We use both effects estimates from MVP lipid traits analysis as well as posterior effects generated by `mashr` package."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "documented-glasgow",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Data used"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7847ca0-e52d-41c7-89ec-ccc3387ebb63",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Reference panel\n",
    "\n",
    "Obtained via `download_1000G()` in `bigsnpr`. \n",
    "\n",
    "Including 503 (mostly unrelated) European individuals and ~1.7M SNPs in common with either HapMap3 or the UK Biobank. Classification of European population can be found at [IGSR](https://www.internationalgenome.org/category/population/). European individuals ID are from [IGSR data portal](https://www.internationalgenome.org/data-portal/sample).\n",
    "\n",
    "### GWAS summary statistics data\n",
    "\n",
    "From MVP. We have the original GWAS summary data as well as multivariate posterior estimate of HDL effects using [mashr](https://github.com/stephenslab/mashr). In brief, we have two versions of summary statistics (effect estimates) for HDL.\n",
    "\n",
    "### Target test data: UK biobank\n",
    "\n",
    "We select randomly from UK Biobank 2000 individuals with covariates and HDL phenotype (medication adjusted, inverse normalized). Their genotypes are extracted. See `UKB.QC.*` PLINK file bundle. \n",
    "\n",
    "### PRS Models\n",
    "\n",
    "Auto model runs the algorithm for 30 different $p$ (the proportion of causal variants) values range from 10e-4 to 0.9, and heritability $h^2$ from LD score regression as initial value.\n",
    "\n",
    "Grid model tries a grid of parameters $p$, ranges from 0 to 1 and three $h^2$ which are 0.7/1/1.4 times of initial $h^2$ estimated by LD score regression."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d175915-5154-478d-a72c-128c2336728a",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Test genotype data preparation\n",
    "\n",
    "Use `awk` select columns in phenotypes file saved to traits file `UKB.hdl.cov` and covariates file `UKB.ind.cov`.\n",
    "In order to merge all bed, bim and fam files,  we use the following command:\n",
    "\n",
    "```\n",
    "for i in {1..22}; do echo ukb_cal_chr$i_v2.bed ukb_snp_chr$i_v2.bim ukb708_cal_chr$i_v2_s488374.fam; done > all_files.txt\n",
    "plink --merge-list all_files.txt --make-bed --out ukbb_merged --threads 30 --memory 100000 \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "465f60f4-6191-40ba-8eb6-eefa096d5fa2",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd(\"~/Documents/PRS_MASH\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1e2aa242-d5dd-43f8-b4b0-b43ac24768eb",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "502536"
      ],
      "text/latex": [
       "502536"
      ],
      "text/markdown": [
       "502536"
      ],
      "text/plain": [
       "[1] 502536"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>502536</li><li>421</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 502536\n",
       "\\item 421\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 502536\n",
       "2. 421\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 502536    421"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>502536</li><li>10</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 502536\n",
       "\\item 10\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 502536\n",
       "2. 10\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 502536     10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>502536</li><li>421</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 502536\n",
       "\\item 421\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 502536\n",
       "2. 421\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 502536    421"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fam_UKB <- read.table(\"UKBB_broad/ukbb_merged.fam\", header = F, stringsAsFactors = F)\n",
    "colnames(fam_UKB)=c(\"FID\",\"IID\",\"paternal.ID\",\"maternal.ID\",\"sex\",\"affection\")\n",
    "covariates <- read.csv(\"7089-27626.csv\", header = T, stringsAsFactors = F)\n",
    "rownames(covariates)=covariates$eid\n",
    "\n",
    "agecov <- read.table(\"UKBB_broad/ukbcov.txt\", sep=\"\\t\", stringsAsFactors = FALSE,header=T)\n",
    "rownames(agecov)=agecov$id\n",
    "i=intersect(agecov$id,covariates$eid)\n",
    "i=as.character(i)\n",
    "length(i)\n",
    "cov2=covariates[i,]\n",
    "agecov=agecov[i,]\n",
    "dim(cov2)\n",
    "dim(agecov)\n",
    "dim(covariates)#HDL is 30760\n",
    "#LDL column name is 30780\n",
    "#TG column names i 30871\n",
    "## TC is 30690\n",
    "## SEc is 31\n",
    "## age is 21011\n",
    "\n",
    "cov=data.frame(\"FID\"=cov2$eid,\"HDL\"=cov2$X30760.0.0,\"LDL\"=cov2$X30780.0.0,\"TG\"=cov2$X30870.0.0,\"TC\"=cov2$X30690.0.0,\"AGE\"=agecov$age,\"SEX\"=agecov$Sex_numeric)\n",
    "\n",
    "suppressMessages(library(tidyverse))\n",
    "\n",
    "set.seed(2021)\n",
    "covariate = cov %>%\n",
    "    drop_na() %>%\n",
    "    filter(FID %in% sample(FID,5000))\n",
    "    \n",
    "covariate = covariate[order(covariate[,1]),]\n",
    "\n",
    "fam_UKB = fam_UKB %>% filter(FID %in% covariate$FID)\n",
    "fam_UKB = fam_UKB[order(covariate[,1]),]\n",
    "colnames(fam_UKB)<-NULL\n",
    "\n",
    "write.table(covariate, file = \"UKBB_broad/UKB.cov\", sep = \" \", \n",
    "            row.names = F, col.names = T)\n",
    "write.table(fam_UKB, file = \"UKBB_broad/UKB.QC.fam\", sep = \" \", \n",
    "            row.names = FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "worth-geometry",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "We only want to focus on selected samples in UKB as target data. Below we extract this subset,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "polyphonic-equity",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mbasic QC filters\u001b[0m: Filter SNPs and select individuals\n",
      "INFO: \u001b[32mbasic QC filters\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mbasic QC filters\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.bed\u001b[0m\n",
      "INFO: Workflow snp_qc (ID=wd7f97a4d9651d36f) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_qc \\\n",
    "    --cwd UKBB_broad \\\n",
    "    --genoFiles UKBB_broad/ukbb_merged.bed \\\n",
    "    --keep-samples UKBB_broad/UKB.QC.fam \\\n",
    "    --geno-filter 0.05 \\\n",
    "    --mind-filter 0.05 \\\n",
    "    --name 5000_subset -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8bd787cc-36ab-4524-a075-22fe30048998",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)\n",
      "Options in effect:\n",
      "  --bfile UKBB_broad/ukbb_merged\n",
      "  --geno 0.05\n",
      "  --hwe 5e-08\n",
      "  --keep UKBB_broad/UKB.QC.fam\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --memory 14400000000\n",
      "  --mind 0.05\n",
      "  --out /home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset\n",
      "  --threads 20\n",
      "\n",
      "Hostname: baymax\n",
      "Working directory: /home/surbut/Documents/PRS_MASH\n",
      "Start time: Fri Jun 11 16:31:19 2021\n",
      "\n",
      "Random number seed: 1623443479\n",
      "64294 MB RAM detected; reserving 14400000000 MB for main workspace.\n",
      "Allocated 61092 MB successfully, after larger attempt(s) failed.\n",
      "784256 variants loaded from .bim file.\n",
      "488377 people (223511 males, 264863 females, 3 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.nosex .\n",
      "--keep: 4955 people remaining.\n",
      "220 people removed due to missing genotype data (--mind).\n",
      "IDs written to\n",
      "/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.irem .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 4735 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.970401.\n",
      "71019 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 3502 variants removed due to Hardy-Weinberg exact test.\n",
      "103253 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "606482 variants and 4735 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.bed +\n",
      "/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.bim +\n",
      "/home/surbut/Documents/PRS_MASH/UKBB_broad/ukbb_merged.5000_subset.fam ...\n",
      "done.\n",
      "\n",
      "End time: Fri Jun 11 16:48:16 2021\n"
     ]
    }
   ],
   "source": [
    "cat UKBB_broad/ukbb_merged.5000_subset.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "6316b4f8-39f9-4638-8ffb-e26d918bc79b",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>4735</li><li>7</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 4735\n",
       "\\item 7\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 4735\n",
       "2. 7\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 4735    7"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cov2=read.table(\"UKBB_broad/UKB.cov\",header = T, stringsAsFactors = F)\n",
    "qcfam=read.table(\"UKBB_broad/ukbb_merged.5000_subset.fam\",header = F, stringsAsFactors = F)\n",
    "\n",
    "i=as.character(qcfam[,2])\n",
    "rownames(cov2)=cov2$\"FID\"\n",
    "\n",
    "cov=cov2[i,]\n",
    "all.equal(cov$FID,qcfam$V2)\n",
    "dim(cov)\n",
    "write.table(cov, file = \"UKBB_broad/UKB.4700.cov\", sep = \" \", \n",
    "            row.names = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "62dce384-4f4b-49a4-b583-e45c6df55707",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "cd ~/Documents/PRS_MASH/UKBB_broad\n",
    "\n",
    "awk '{print $6, $7}' UKB.4700.cov > UKB.ind.cov\n",
    "awk '{print $3}' UKB.4700.cov > UKB.ldl.cov\n",
    "awk '{print $2}' UKB.4700.cov > UKB.hdl.cov\n",
    "awk '{print $4}' UKB.4700.cov > UKB.tg.cov\n",
    "awk '{print $5}' UKB.4700.cov > UKB.tc.cov\n",
    "\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dried-episode",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "At this point files on the disk should be:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "alert-hunger",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[01;34m.\u001b[00m\n",
      "├── \u001b[01;34m1000G.EUR\u001b[00m\n",
      "│   ├── 1000G.EUR.bed\n",
      "│   ├── 1000G.EUR.bim\n",
      "│   └── 1000G.EUR.fam\n",
      "├── 7089-27626.csv\n",
      "├── ldpred.html\n",
      "├── \u001b[01;36mldpred.ipynb\u001b[00m -> /home/surbut/Projects/bioworkflows/ldpred/ldpred.ipynb\n",
      "├── \u001b[01;34mmvpdata\u001b[00m\n",
      "│   ├── chr_pos_allele2_lfsr.txt\n",
      "│   ├── gwas_hdl.rds\n",
      "│   ├── gwas_ldl.rds\n",
      "│   ├── gwas_tc.rds\n",
      "│   ├── gwas_tg.rds\n",
      "│   ├── mash_hdl.rds\n",
      "│   ├── mash_ldl.rds\n",
      "│   ├── mash_tc.rds\n",
      "│   ├── mash_tg.rds\n",
      "│   ├── Merged_MVP_Full_se_raw.txt\n",
      "│   ├── MVP_all_beta_posterior_beta.txt\n",
      "│   ├── posterior_beta_se.txt\n",
      "│   └── zmash_raw_univariate_MVP.rds\n",
      "├── \u001b[01;34mmvp_gwas\u001b[00m\n",
      "│   ├── 1000G.EUR.mvp_gwas.bed\n",
      "│   ├── 1000G.EUR.mvp_gwas.bim\n",
      "│   ├── 1000G.EUR.mvp_gwas.fam\n",
      "│   ├── 1000G.EUR.mvp_gwas.log\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.bed\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.bim\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.bk\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.fam\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.log\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.rds\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.stderr\n",
      "│   ├── 1000G.EUR.mvp_gwas.snp_intersect.extracted.stdout\n",
      "│   ├── 1000G.EUR.mvp_gwas.stdout\n",
      "│   ├── chr10.OMNI.interpolated_genetic_map\n",
      "│   ├── chr11.OMNI.interpolated_genetic_map\n",
      "│   ├── chr12.OMNI.interpolated_genetic_map\n",
      "│   ├── chr13.OMNI.interpolated_genetic_map\n",
      "│   ├── chr14.OMNI.interpolated_genetic_map\n",
      "│   ├── chr15.OMNI.interpolated_genetic_map\n",
      "│   ├── chr16.OMNI.interpolated_genetic_map\n",
      "│   ├── chr17.OMNI.interpolated_genetic_map\n",
      "│   ├── chr18.OMNI.interpolated_genetic_map\n",
      "│   ├── chr19.OMNI.interpolated_genetic_map\n",
      "│   ├── chr1.OMNI.interpolated_genetic_map\n",
      "│   ├── chr20.OMNI.interpolated_genetic_map\n",
      "│   ├── chr21.OMNI.interpolated_genetic_map\n",
      "│   ├── chr22.OMNI.interpolated_genetic_map\n",
      "│   ├── chr2.OMNI.interpolated_genetic_map\n",
      "│   ├── chr3.OMNI.interpolated_genetic_map\n",
      "│   ├── chr4.OMNI.interpolated_genetic_map\n",
      "│   ├── chr5.OMNI.interpolated_genetic_map\n",
      "│   ├── chr6.OMNI.interpolated_genetic_map\n",
      "│   ├── chr7.OMNI.interpolated_genetic_map\n",
      "│   ├── chr8.OMNI.interpolated_genetic_map\n",
      "│   ├── chr9.OMNI.interpolated_genetic_map\n",
      "│   ├── gwas_hdl.intersect.rds\n",
      "│   ├── gwas_hdl.intersect.snplist\n",
      "│   ├── gwas_hdl.intersect.stdout\n",
      "│   ├── gwas_hdl.snp_matched.auto_prs.stderr\n",
      "│   ├── gwas_hdl.snp_matched.auto_prs.stdout\n",
      "│   ├── gwas_hdl.snp_matched.inf_prs.rds\n",
      "│   ├── gwas_hdl.snp_matched.inf_prs.stderr\n",
      "│   ├── gwas_hdl.snp_matched.inf_prs.stdout\n",
      "│   ├── gwas_hdl.snp_matched.ld.rds\n",
      "│   ├── gwas_hdl.snp_matched.ld.stderr\n",
      "│   ├── gwas_hdl.snp_matched.rds\n",
      "│   ├── gwas_hdl.snp_matched.snplist\n",
      "│   ├── gwas_hdl.snp_matched.stderr\n",
      "│   ├── gwas_ldl.intersect.rds\n",
      "│   ├── gwas_ldl.intersect.snplist\n",
      "│   ├── gwas_ldl.intersect.stdout\n",
      "│   ├── gwas_ldl.snp_matched.inf_prs.rds\n",
      "│   ├── gwas_ldl.snp_matched.inf_prs.stderr\n",
      "│   ├── gwas_ldl.snp_matched.inf_prs.stdout\n",
      "│   ├── gwas_ldl.snp_matched.ld.rds\n",
      "│   ├── gwas_ldl.snp_matched.ld.stderr\n",
      "│   ├── \u001b[01;35mgwas_ldl.snp_matched.qc.png\u001b[00m\n",
      "│   ├── gwas_ldl.snp_matched.qc.rds\n",
      "│   ├── gwas_ldl.snp_matched.qc.snplist\n",
      "│   ├── gwas_ldl.snp_matched.qc.stderr\n",
      "│   ├── gwas_ldl.snp_matched.qc.stdout\n",
      "│   ├── gwas_ldl.snp_matched.rds\n",
      "│   ├── gwas_ldl.snp_matched.snplist\n",
      "│   ├── gwas_ldl.snp_matched.stderr\n",
      "│   ├── gwas_tg.intersect.rds\n",
      "│   ├── gwas_tg.intersect.snplist\n",
      "│   ├── gwas_tg.intersect.stdout\n",
      "│   ├── \u001b[01;34mld-cache\u001b[00m\n",
      "│   │   ├── filea9eb7da8e69.sbk\n",
      "│   │   ├── filec69c78a6fb03.sbk\n",
      "│   │   └── filed06c176aa005.sbk\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.bed\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.bim\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.bk\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.fam\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.log\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.rds\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.stderr\n",
      "│   ├── UKB.2000_subset.snp_intersect.extracted.stdout\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.bed\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.bim\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.bk\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.fam\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.log\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.rds\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.stderr\n",
      "│   ├── ukbb_merged.5000_subset.snp_intersect.extracted.stdout\n",
      "│   ├── UKB.hdl.gwas_hdl.snp_matched.inf_prs.stderr\n",
      "│   ├── UKB.ldl.baseline.pdf\n",
      "│   ├── UKB.ldl.baseline.rds\n",
      "│   ├── UKB.ldl.baseline.stderr\n",
      "│   ├── UKB.ldl.baseline.stdout\n",
      "│   ├── UKB.ldl.gwas_ldl.snp_matched.inf_prs.pdf\n",
      "│   ├── UKB.ldl.gwas_ldl.snp_matched.inf_prs.rds\n",
      "│   ├── UKB.ldl.gwas_ldl.snp_matched.inf_prs.stderr\n",
      "│   └── UKB.ldl.gwas_ldl.snp_matched.inf_prs.stdout\n",
      "├── plink.log\n",
      "├── \u001b[01;34mUKBB\u001b[00m\n",
      "│   ├── \u001b[01;32mlipid_demographics.csv\u001b[00m\n",
      "│   └── \u001b[01;32mREAD_ME_fam_files.txt\u001b[00m\n",
      "├── \u001b[01;34mUKBB_broad\u001b[00m\n",
      "│   ├── all_files.txt\n",
      "│   ├── all_files.txt~\n",
      "│   ├── big_ukb_file.txt\n",
      "│   ├── \u001b[01;34mnewfamdir\u001b[00m\n",
      "│   │   ├── mymerged.bed\n",
      "│   │   ├── mymerged.bim\n",
      "│   │   └── mymerged.fam\n",
      "│   ├── pheno_ukbb.rdata\n",
      "│   ├── UKB.4700.cov\n",
      "│   ├── UKB.4747.cov\n",
      "│   ├── ukb708_cal_chr10_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr11_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr12_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr13_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr14_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr15_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr16_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr17_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr18_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr19_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr1_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr20_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr21_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr22_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr2_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr3_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr4_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr5_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr6_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr7_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr8_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chr9_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chrMT_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chrX_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chrXY_v2_s488374.fam\n",
      "│   ├── ukb708_cal_chrY_v2_s488374.fam\n",
      "│   ├── ukbb_merged.2000_subset.irem\n",
      "│   ├── ukbb_merged.2000_subset.log\n",
      "│   ├── ukbb_merged.2000_subset.nosex\n",
      "│   ├── ukbb_merged.2000_subset.stderr\n",
      "│   ├── ukbb_merged.2000_subset.stdout\n",
      "│   ├── ukbb_merged.5000_subset.bed\n",
      "│   ├── ukbb_merged.5000_subset.bim\n",
      "│   ├── ukbb_merged.5000_subset.fam\n",
      "│   ├── ukbb_merged.5000_subset.irem\n",
      "│   ├── ukbb_merged.5000_subset.log\n",
      "│   ├── ukbb_merged.5000_subset.nosex\n",
      "│   ├── ukbb_merged.5000_subset.stdout\n",
      "│   ├── ukbb_merged.bed\n",
      "│   ├── ukbb_merged.bim\n",
      "│   ├── ukbb_merged.fam\n",
      "│   ├── ukbb_merged.log\n",
      "│   ├── ukbb_merged.nosex\n",
      "│   ├── \u001b[01;34mukbb_merged.UKBB_broad\u001b[00m\n",
      "│   │   ├── ukbb_merged_qced.irem\n",
      "│   │   ├── ukbb_merged_qced.log\n",
      "│   │   ├── ukbb_merged_qced.nosex\n",
      "│   │   ├── ukbb_merged_qced.stderr\n",
      "│   │   └── ukbb_merged_qced.stdout\n",
      "│   ├── ukb_cal_chr10_v2.bed\n",
      "│   ├── ukb_cal_chr11_v2.bed\n",
      "│   ├── ukb_cal_chr12_v2.bed\n",
      "│   ├── ukb_cal_chr13_v2.bed\n",
      "│   ├── ukb_cal_chr14_v2.bed\n",
      "│   ├── ukb_cal_chr15_v2.bed\n",
      "│   ├── ukb_cal_chr16_v2.bed\n",
      "│   ├── ukb_cal_chr17_v2.bed\n",
      "│   ├── ukb_cal_chr18_v2.bed\n",
      "│   ├── ukb_cal_chr19_v2.bed\n",
      "│   ├── ukb_cal_chr1_v2.bed\n",
      "│   ├── ukb_cal_chr20_v2.bed\n",
      "│   ├── ukb_cal_chr21_v2.bed\n",
      "│   ├── ukb_cal_chr22_v2.bed\n",
      "│   ├── ukb_cal_chr2_v2.bed\n",
      "│   ├── ukb_cal_chr3_v2.bed\n",
      "│   ├── ukb_cal_chr4_v2.bed\n",
      "│   ├── ukb_cal_chr5_v2.bed\n",
      "│   ├── ukb_cal_chr6_v2.bed\n",
      "│   ├── ukb_cal_chr7_v2.bed\n",
      "│   ├── ukb_cal_chr8_v2.bed\n",
      "│   ├── ukb_cal_chr9_v2.bed\n",
      "│   ├── ukb_cal_chrMT_v2.bed\n",
      "│   ├── ukb_cal_chrX_v2.bed\n",
      "│   ├── ukb_cal_chrXY_v2.bed\n",
      "│   ├── ukb_cal_chrY_v2.bed\n",
      "│   ├── UKB.cov\n",
      "│   ├── ukbcov.txt\n",
      "│   ├── UKB.hdl.cov\n",
      "│   ├── UKB.ind.cov\n",
      "│   ├── UKB.ldl.cov\n",
      "│   ├── UKB.QC.fam\n",
      "│   ├── ukb_snp_chr10_v2.bim\n",
      "│   ├── ukb_snp_chr11_v2.bim\n",
      "│   ├── ukb_snp_chr12_v2.bim\n",
      "│   ├── ukb_snp_chr13_v2.bim\n",
      "│   ├── ukb_snp_chr14_v2.bim\n",
      "│   ├── ukb_snp_chr15_v2.bim\n",
      "│   ├── ukb_snp_chr16_v2.bim\n",
      "│   ├── ukb_snp_chr17_v2.bim\n",
      "│   ├── ukb_snp_chr18_v2.bim\n",
      "│   ├── ukb_snp_chr19_v2.bim\n",
      "│   ├── ukb_snp_chr1_v2.bim\n",
      "│   ├── ukb_snp_chr20_v2.bim\n",
      "│   ├── ukb_snp_chr21_v2.bim\n",
      "│   ├── ukb_snp_chr22_v2.bim\n",
      "│   ├── ukb_snp_chr2_v2.bim\n",
      "│   ├── ukb_snp_chr3_v2.bim\n",
      "│   ├── ukb_snp_chr4_v2.bim\n",
      "│   ├── ukb_snp_chr5_v2.bim\n",
      "│   ├── ukb_snp_chr6_v2.bim\n",
      "│   ├── ukb_snp_chr7_v2.bim\n",
      "│   ├── ukb_snp_chr8_v2.bim\n",
      "│   ├── ukb_snp_chr9_v2.bim\n",
      "│   ├── ukb_snp_chrMT_v2.bim\n",
      "│   ├── ukb_snp_chrX_v2.bim\n",
      "│   ├── ukb_snp_chrXY_v2.bim\n",
      "│   ├── ukb_snp_chrY_v2.bim\n",
      "│   ├── UKB.tc.cov\n",
      "│   └── UKB.tg.cov\n",
      "└── \u001b[01;34mukbiobank\u001b[00m\n",
      "    ├── UKB.2000_subset.bed\n",
      "    ├── UKB.2000_subset.bim\n",
      "    ├── UKB.2000_subset.fam\n",
      "    ├── UKB.cov\n",
      "    ├── UKB.hdl.cov\n",
      "    ├── UKB.ind.cov\n",
      "    └── UKB.QC.fam\n",
      "\n",
      "9 directories, 241 files\n"
     ]
    }
   ],
   "source": [
    "tree"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "accepting-pepper",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Analysis of MVP GWAS data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "rolled-ending",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 1: QC on reference panel\n",
    "\n",
    "Here we assume the target data QC has been already performed. We perform here QC for reference panel,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "israeli-costa",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "work_dir=mvp_gwas\n",
    "cd ~/Documents/PRS_MASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "stupid-preservation",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb snp_qc \\\n",
    "    --cwd $work_dir \\\n",
    "    --genoFiles 1000G.EUR/1000G.EUR.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "foreign-analyst",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 2: Intersect SNPs among summary stats, reference panel and target data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "4b00aa7b-7c22-4a54-b717-4b4f66bc1cc1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "work_dir=mvp_gwas\n",
    "lipid=hdl\n",
    "data=gwas\n",
    "cd ~/Documents/PRS_MASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "invalid-channels",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32msnp_intersect_1\u001b[0m: SNP intersect of summary stats and genotype data\n",
      "INFO: \u001b[32msnp_intersect_1\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32msnp_intersect_1\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.intersect.rds /home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.intersect.snplist\u001b[0m\n",
      "INFO: Running \u001b[32msnp_intersect_2\u001b[0m: \n",
      "INFO: \u001b[32msnp_intersect_2\u001b[0m is \u001b[32mcompleted\u001b[0m (pending nested workflow).\n",
      "INFO: Running \u001b[32mpreprocess_1\u001b[0m: Filter SNPs and select individuals\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpreprocess_1\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bed /home/surbut/Documents/PRS_MASH/mvp_gwas/ukbb_merged.5000_subset.snp_intersect.extracted.bed in 2 groups\u001b[0m\n",
      "INFO: Running \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m: \n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mconvert PLNIK to bigsnpr format with missing data mean imputed\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bk /home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.rds... (4 items in 2 groups)\u001b[0m\n",
      "INFO: \u001b[32msnp_intersect_2\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.bed /home/surbut/Documents/PRS_MASH/mvp_gwas/1000G.EUR.mvp_gwas.snp_intersect.extracted.rds... (4 items)\u001b[0m\n",
      "INFO: Workflow snp_intersect (ID=wffa5f5d1d8100e11) is executed successfully with 4 completed steps and 6 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_intersect \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss mvpdata/$data\"_\"$lipid.rds \\\n",
    "    --genoFiles $work_dir/1000G.EUR.$work_dir.bed UKBB_broad/ukbb_merged.5000_subset.bed -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "committed-arrival",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tail: cannot open 'mvp_gwas/hdl.intersect.stdout' for reading: No such file or directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "tail -1 $work_dir/$data\"_\"$lipid.intersect.stdout"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "integral-ranch",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 3: Harmonize alleles for shared SNPs\n",
    "\n",
    "To handle major/minor allele, strand flips and consequently possible flips in sign for summary statistics."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dried-mechanics",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32msnp_match\u001b[0m: \n",
      "INFO: \u001b[32msnp_match\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32msnp_match\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.snp_matched.rds /home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.snp_matched.snplist\u001b[0m\n",
      "INFO: Workflow snp_match (ID=w8255e53608faedab) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb snp_match \\\n",
    "    --cwd $work_dir \\\n",
    "    --reference_geno $work_dir/1000G.EUR.$work_dir.snp_intersect.extracted.rds \\\n",
    "    --ss mvpdata/$data\"_\"$lipid.rds -s force"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "considered-infection",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 4: Calculate LD matrix and fit LDSC model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "several-armenia",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mldsc\u001b[0m: \n",
      "INFO: \u001b[32mldsc\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mldsc\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.snp_matched.ld.rds\u001b[0m\n",
      "INFO: Workflow ldsc (ID=w074f16b85a5b6dd6) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb ldsc \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --reference-geno $work_dir/1000G.EUR.$work_dir.snp_intersect.extracted.rds -s force"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "returning-investigator",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 6: Estimate posterior effect sizes and PRS\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "political-stack",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "For original data,"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "modular-midnight",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32minf_prs\u001b[0m: \n",
      "INFO: \u001b[32minf_prs\u001b[0m is \u001b[32mcompleted\u001b[0m (pending nested workflow).\n",
      "INFO: Running \u001b[32mprs_core\u001b[0m: \n",
      "INFO: \u001b[32mprs_core\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mprs_core\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: \u001b[32minf_prs\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/gwas_hdl.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: Workflow inf_prs (ID=wbdd86af79f964fbe) is executed successfully with 2 completed steps.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb inf_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data_$lipid.snp_matched.ld.rds -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "interpreted-temple",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"422669 SNPs are used for PRS calculations\"\n"
     ]
    }
   ],
   "source": [
    "tail -1 mvp_gwas/$data\"_\"$lipid.snp_matched.inf_prs.stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "running-highlight",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb auto_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data\"_\"$lipid.snp_matched.ld.rds -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "worst-hungarian",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "sos run ldpred.ipynb grid_prs \\\n",
    "    --cwd $work_dir \\\n",
    "    --ss $work_dir/$data\"_\"$lipid.snp_matched.rds \\\n",
    "    --target-geno $work_dir/ukbb_merged.5000_subset.snp_intersect.extracted.rds \\\n",
    "    --ldsc $work_dir/$data\"_\"$lipid.snp_matched.ld.rds \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "nonprofit-haven",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Step 7: predict phenotypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "female-uniform",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Baseline model: Traits ~ Sex + Age"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "8804f629-e8ca-4737-8cd6-b02f57919280",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ldl\n"
     ]
    }
   ],
   "source": [
    "echo $lipid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "pharmaceutical-cherry",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mpred_eval\u001b[0m: \n",
      "INFO: \u001b[32mpred_eval\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpred_eval\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/UKB.ldl.baseline.rds\u001b[0m\n",
      "INFO: Workflow pred_eval (ID=w94cff0257f2465bb) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "fabulous-flour",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = ., data = dat[train.ind, ])\n",
       "\n",
       "Residuals:\n",
       "    Min      1Q  Median      3Q     Max \n",
       "-2.2741 -0.6402 -0.0483  0.5696  4.1151 \n",
       "\n",
       "Coefficients:\n",
       "              Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  3.5341602  0.0988638  35.748   <2e-16 ***\n",
       "AGE          0.0007752  0.0017223   0.450    0.653    \n",
       "SEX         -0.0221843  0.0285231  -0.778    0.437    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 0.8712 on 3794 degrees of freedom\n",
       "Multiple R-squared:  0.000207,\tAdjusted R-squared:  -0.0003201 \n",
       "F-statistic: 0.3927 on 2 and 3794 DF,  p-value: 0.6753\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>model</th><th scope=col>R2</th><th scope=col>MSE</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>model</td><td>-0.00032</td><td>0.81536</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " model & R2 & MSE\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t model & -0.00032 & 0.81536\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| model &lt;chr&gt; | R2 &lt;dbl&gt; | MSE &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| model | -0.00032 | 0.81536 |\n",
       "\n"
      ],
      "text/plain": [
       "  model R2       MSE    \n",
       "1 model -0.00032 0.81536"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lipid=\"hdl\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".baseline.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "different-tragedy",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Inf/grid/auto model: Traits ~ Sex + Age + PRS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "assumed-moment",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mpred_eval\u001b[0m: \n",
      "INFO: \u001b[32mpred_eval\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mpred_eval\u001b[0m output:   \u001b[32m/home/surbut/Documents/PRS_MASH/mvp_gwas/UKB.ldl.gwas_ldl.snp_matched.inf_prs.rds\u001b[0m\n",
      "INFO: Workflow pred_eval (ID=w0855c9540afce4c1) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/data\"_\"$lipid.snp_matched.inf_prs.rds \\\n",
    "    --phenoFile UKBB_broad/UKB.$lipid.cov \\\n",
    "    --covFile UKBB_broad/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "sharing-report",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = ., data = dat[train.ind, ])\n",
       "\n",
       "Residuals:\n",
       "    Min      1Q  Median      3Q     Max \n",
       "-2.3695 -0.6448 -0.0338  0.5562  3.8091 \n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  3.501340   0.097526  35.901   <2e-16 ***\n",
       "AGE          0.001072   0.001698   0.631    0.528    \n",
       "SEX         -0.017764   0.028126  -0.632    0.528    \n",
       "PRS         -0.569875   0.054384 -10.479   <2e-16 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 0.859 on 3793 degrees of freedom\n",
       "Multiple R-squared:  0.02834,\tAdjusted R-squared:  0.02757 \n",
       "F-statistic: 36.87 on 3 and 3793 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>model</th><th scope=col>R2</th><th scope=col>MSE</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>model.inf_prs</td><td>0.02757</td><td>0.78358</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " model & R2 & MSE\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t model.inf\\_prs & 0.02757 & 0.78358\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 3\n",
       "\n",
       "| model &lt;chr&gt; | R2 &lt;dbl&gt; | MSE &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| model.inf_prs | 0.02757 | 0.78358 |\n",
       "\n"
      ],
      "text/plain": [
       "  model         R2      MSE    \n",
       "1 model.inf_prs 0.02757 0.78358"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lipid=\"hdl\"\n",
    "data=\"gwas\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.inf_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "least-cinema",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/$data\"_\"$lipid.snp_matched.grid_prs.rds \\\n",
    "    --phenoFile ukbiobank/UKB.$lipid.cov \\\n",
    "    --covFile ukbiobank/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8fa95a97-824c-432e-9221-d52f227aecb3",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "lipid=\"hdl\"\n",
    "data=\"gwas\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.grid_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "certified-marriage",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "\n",
    "sos run ldpred.ipynb pred_eval \\\n",
    "    --cwd $work_dir \\\n",
    "    --prs $work_dir/$data\"_\"$lipid.snp_matched.auto_prs.rds \\\n",
    "    --phenoFile ukbiobank/UKB.$lipid.cov \\\n",
    "    --covFile ukbiobank/UKB.ind.cov \\\n",
    "    --response continuous -s force"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "regular-parking",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": [
    "lipid=\"hdl\"\n",
    "data=\"gwas\"\n",
    "res = readRDS(paste0(\"mvp_gwas/UKB.\",lipid,\".\",data,\"_\",lipid,\".snp_matched.auto_prs.rds\"))\n",
    "summary(res$fitted)\n",
    "res$summary"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.22.5"
  },
  "toc-autonumbering": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
