<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sarah Urbut" />


<title>Clumping and Thresholding Windows by proxy</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lipids_mvp</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Clumping and Thresholding Windows by proxy</h1>
<h4 class="author">Sarah Urbut</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lipids_mvp/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsurbutlipidsmvptreecfb45a20c4a526ed2d2dc1c9044d17591ab5f87ctargetblankcfb45a2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/surbut/lipids_mvp/tree/cfb45a20c4a526ed2d2dc1c9044d17591ab5f87c" target="_blank">cfb45a2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsurbutlipidsmvptreecfb45a20c4a526ed2d2dc1c9044d17591ab5f87ctargetblankcfb45a2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/surbut/lipids_mvp/tree/cfb45a20c4a526ed2d2dc1c9044d17591ab5f87c" target="_blank">cfb45a2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/analysingrepro.nb.html
    Ignored:    analysis/ct.nb.html
    Ignored:    analysis/looking_at_significance.nb.html
    Ignored:    analysis/looking_at_significance_cache/
    Ignored:    analysis/simualtions.nb.html
    Ignored:    analysis/simulations.nb.html
    Ignored:    analysis/workingwithdata.nb.html
    Ignored:    analysis/workingwithdata_cache/
    Ignored:    scripts/.Rapp.history

Untracked files:
    Untracked:  analysis/C+T Doc.rmd.md
    Untracked:  analysis/venn_local.Rmd
    Untracked:  scripts/preparingmashmodel_local.Rmd

Unstaged changes:
    Deleted:    analysis/eCDF.Rmd
    Deleted:    analysis/venn.Rmd
    Deleted:    analysis/workingwithdata.Rmd
    Modified:   scripts/preparingmashdat.Rmd
    Deleted:    scripts/preparingmashmodel.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ct.rmd</code>) and HTML (<code>docs/ct.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/surbut/lipids_mvp/blob/cfb45a20c4a526ed2d2dc1c9044d17591ab5f87c/analysis/ct.rmd" target="_blank">cfb45a2</a>
</td>
<td>
Sarah Urbut
</td>
<td>
2020-09-01
</td>
<td>
wflow_publish(“../../lipids_mvp/analysis/ct.rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The C+T polygenic score is defined as the sum of allele counts (genotypes), weighted by estimated effect sizes ob tained from genome-wide association studies, where two filtering steps have been applied (Wray et al. 2007; Purcell et al. 2009; Dudbridge 2013; Wray et al. 2014; Euesden et al. 2014; Chatterjee et al. 2016).</p>
<p>More precisely, the variants are first clumped (C) so that only variants that are weakly correlated with one another are retained.</p>
<p>Clumping selects the most significant variant iteratively, computes correlation between this index variant and nearby variants within some genetic distance wc, and removes all the nearby variants that are correlated with this index variant beyond 2 a particular value r_{2}.</p>
<p>Thresholding (T) consists in removing variants with a p-value larger than a chosen level of significance <span class="math inline">\((p &gt; p_{T})\)</span>. Both steps, clumping and thresholding, represent a statistical compromise between signal and noise. The clumping step prunes redundant correlated effects caused by linkage disequilibrium (LD) between variants. However, this procedure may also remove independently predictive variants in LD. Similarly, thresholding must balance between including truly predictive variants and reducing noise in the score by excluding null effects.</p>
<p>When applying C+T, one has three hyper-parameters to select, namely the squared correlation threshold <span class="math inline">\(r_{2}\)</span> and the window size <span class="math inline">\(w_{c}\)</span> of clumping, along with the p value threshold <span class="math inline">\(p_{T}\)</span>. Usually, C+T users assign default values for clumping, such as <span class="math inline">\(r_{c}\)</span> of 0.1 (default of PRSice), 0.2 or 0.5 (default of PLINK), and <span class="math inline">\(w_{c}\)</span> of 250 kb (default of PRSice).</p>
<p>An example of this output is from PLINK below:</p>
<pre><code> CHR    F               SNP         BP        P    TOTAL   NSIG    S05    S01   S001  S0001    SP2
   1    1    1:55500429:A:G   55500429          0        1      0      0      0      0      1 1:55502137:C:G(1)
   1    1    1:55513061:T:C   55513061          0        2      0      0      0      0      2 1:55511995:A:G(1),1:55516508:A:G(1)
   1    1    1:55522141:A:G   55522141          0        3      0      0      0      0      3 1:55518467:A:G(1),1:55521242:T:G(1),1:55557540:T:C(1)
   1    1    1:55526428:T:G   55526428          0        4      0      0      0      0      4 1:55543079:T:C(1),1:55552273:A:G(1),1:55605395:T:C(1),1:55638546:T:C(1)</code></pre>
<p>where the column labeled SNP represents the index SNP that is both <span class="math inline">\(\leq r_{c}\)</span> and chosen to satisfy the index <strong>lfsr</strong> threshold chose.</p>
<p>Plink proceeds with the following structure: There are four main parameters that determine the level of clumping, listed here in terms of the command flag used to change them and their default values:</p>
<pre><code>   --clump-p1 0.0001          Significance threshold for index SNPs
 
   --clump-p2 0.01            Secondary significance threshold for clumped SNPs

   --clump-r2 0.50            LD threshold for clumping

   --clump-kb 250             Physical distance threshold for clumping
 </code></pre>
<p>The clumping procedure takes all SNPs that are significant at threshold p1 (or lfsr in our case). that have not already been clumped (denoting these as index SNPs) and forms clumps of all other SNPs that are within a certain kb distance from the index SNP (default 250kb) and that are in linkage disequilibrium with the index SNP, based on an r-squared threshold (default 0.50). These SNPs are then subsetted based on the result for that SNP, as illustrated in the SP2 column above.</p>
<div id="practical-application" class="section level2">
<h2>Practical Application</h2>
<p>In our application, we filter in two additional ways:</p>
<p><strong>1</strong> Only those variants that exist both in MVP and in 1000 genomes (1kg) are retained for clumping and thresholding</p>
<p><strong>2</strong> Only the independent <code>index</code> snp is used for potential scoring.</p>
<p>The potential loss of useful infomration thus occurs in two steps, as we choose only SNPs that are genotyped and present in all 650K ’proportional individuals and the MVP individuals</p>
<p><strong>1</strong> In limiting our index variants to only those that first satisfy the index variant lfsr threshold p1 above, we reduce the possible LD blocks identified.</p>
<p><strong>2</strong> We risk losing the strongest signal in the window if it is not captured by both MVP and 1kg.</p>
</div>
<div id="proposal" class="section level2">
<h2>Proposal</h2>
<p>In order to use the genotype information from 1kg but also the depth of our MVP coverage, I propose a two fold thresholding step, <span class="math inline">\(t_{0}\)</span>.</p>
<p><strong>1</strong> First, using the 1kg data that overlaps MVP (i.e., the ~4M variants), <code>clump</code> with a relaxed lfsr threshold <span class="math inline">\(t_{0}\)</span> of, suppose 0.5, or even 0.7. The goal here is not to capture true index SNPs, but really to define independent LD blocks, for which we will later threshold MVP (and not 1kg variants). Perform the usual steps of removing all variants within some distance <span class="math inline">\(w_{c}\)</span> that exceed an <span class="math inline">\(r^{2}\)</span> threshold. It is not necessary to retain the <code>clumped</code> SNPs (column p2) so to ease the computational burden p2 in this case can be very low (i.e. 1e-7).</p>
<p><strong>2</strong> After defending these soft-index SNPS, define a new window A’ and B’ with boundaries 250 kb from soft index SNP. Choose the minimum MVP SNP in this window such that it satisfies the true lfsr threshold we are interested in. This can be significantly stringent, but ideally given the depth of coverage in MVP, we will identify both more independent signals and estimates that are stronger in magnitude in precision.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Privé, Florian et al. The American Journal of Human Genetics, Volume 105, Issue 6, 1213 - 1221</p>
<p>Plink, Clumping Documentation, <a href="https://zzz.bwh.harvard.edu/plink/clump.shtml" class="uri">https://zzz.bwh.harvard.edu/plink/clump.shtml</a>.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
